pkg/support/dump_test.go

```diff
@@ -38,3 +138,111 @@ func TestParseFileName(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, ts.String(), "2021-08-17 09:47:58 +0000 UTC")
 }
+
+func TestTimestampFilter(t *testing.T) {
+	ts := timestampFilter("2m")
+	currentTime := time.Now()
+	require.NotNil(t, ts)
+	expectedTime := currentTime.Add(-2 * time.Minute)
+	assert.WithinDuration(t, expectedTime, *ts, 10*time.Second)
+}
+
+func TestWriteFile(t *testing.T) {
+	basedir := "dir1/dir2"
+	resource := "networkpolicies"
+	data := []byte("antct -oyaml get networkpolicies")
+
+	fs := afero.NewMemMapFs()
+	err := writeFile(fs, filepath.Join(basedir, resource), resource, data)
+	require.NoError(t, err)
+
+	fileContents, err := afero.ReadFile(fs, filepath.Join(basedir, resource))
+	require.NoError(t, err)
+	require.Equal(t, data, fileContents)
+}
+
+func TestWriteYAMLFile(t *testing.T) {
+	basedir := "dir1/dir2"
+	resource := "pod.yaml"
+	yamlData := `apiVersion: v1
+kind: Pod
+metadata:
+  name: my-pod`
+
+	fs := afero.NewMemMapFs()
+	err := writeYAMLFile(fs, filepath.Join(basedir, resource), resource, yamlData)
```

Comment:
this test is not representative of how the function is actually being used. The last parameter (named `data`) is typically a Golang object or pointer to an object. Here is a better version:
```go
func TestWriteYAMLFile(t *testing.T) {
	type testData struct {
		X string
		Y int
	}
	data := testData{
		X: "foo",
		Y: 123,
	}
	resource := "test-data"
	path := filepath.Join(basedir, resource)
	fs := afero.NewMemMapFs()
	err := writeYAMLFile(fs, path, resource, &data)
	require.NoError(t, err)

	fileContents, err := afero.ReadFile(fs, path)
	require.NoError(t, err)
	var fileData testData
	require.NoError(t, yaml.Unmarshal(fileContents, &fileData))
	assert.Equal(t, data, fileData)
}
```